File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/workload/interoperationtimingstype/InterOperationsTimingType.scala
package cloud.benchflow.dsl.definition.workload.interoperationtimingstype

/**
 * @author Jesper Findahl (jesper.findahl@gmail.com)
 *         created on 2017-05-23
 */
object InterOperationsTimingType extends Enumeration {
  type InterOperationsTimingType = Value
  val NegativeExponential = Value("negative-exponential")
  val Uniform = Value("uniform")
  val FixedTime = Value("fixed-time")
}

File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/workload/interoperationtimingstype/InterOperationsTimingType.java
package cloud.benchflow.dsl.definition.workload.interoperationtimingstype;

/**
 * @author Jesper Findahl (jesper.findahl@gmail.com)
 *         created on 2017-06-23
 */
public enum InterOperationsTimingType {
  NEGATIVE_EXPONENTIAL("negative-exponential"),
  UNIFORM("uniform"),
  FIXED_TIME("fixed-time")
  ;


  private final String stringValue;

  InterOperationsTimingType(String stringValue) {
    this.stringValue = stringValue;
  }

  @Override
  public String toString() {
    return stringValue;
  }
}
